hyperspy/drawing/widgets.py

```diff
@@ -67,40 +71,48 @@ def set_on(self, value):
                     if self.patch in container:
                         container.remove(self.patch)
                 self.disconnect(self.ax)
+                self.ax = None
             self.__is_on = value
             try:
                 self.ax.figure.canvas.draw()
             except:  # figure does not exist
                 pass
-            else:
-                self.ax = None
 
-    def set_patch(self):
+    def _set_patch(self):
         pass
         # Must be provided by the subclass
 
-    def add_patch_to(self, ax):
-        self.set_patch()
+    def _add_patch_to(self, ax):
+        self._set_patch()
         ax.add_artist(self.patch)
         self.patch.set_animated(hasattr(ax, 'hspy_fig'))
 
-    def add_axes(self, ax):
+    def set_axis(self, ax):
```

Comment:
I agree that `add_axes` was misleading. However, `ax` should be a MPL Axes instance, so `set_axis` doesn't feel fully right either. What about `set_mpl_axes` or `set_mpl_ax`?